src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -496,6 +499,9 @@ void BackupCoordinationStageSync::watchingThread()
             tryLogCurrentException(log, "Caught exception while watching");
         }
 
+        if (should_stop())
+            return;
```

Comment:
This is to help a cancelled backup to stop working faster, that must help `test_backup_restore_on_cluster/test_cancel_backup.py::test_cancel_backup`.